### ISSUE TITLE
Playlist Sorting 

### ISSUE THREAD

**Issue by ListSort** [TeamNewPipe/NewPipe#3263]
If users could sort their playlists by video length or title or channel name this would make it easy to see and be able to remove duplicate videos from playlists.

It would also be great when adding a video to a playlist to be able to see all the playlists it is already in... to prevent duplicates.

**Comment by Stypox** [comment]
See #1473 (maybe in combination with #3037) are better solutions to this problem, imho

**Comment by ListSort** [comment]
I saw #1473 prior to posting this. It was suggested nearly 2 years ago now.

That is why I made this suggestion. I believe this would be much easier to implement. If it were a goal it would be achieved much quicker and make the program much more user friendly and more popular. Faster return for less effort. Imho.

**Comment by ListSort** [comment]
I've been using newpipe nearly 2 weeks now.

The 2 things I notice & miss most switching from youtube app to newpipe are 

1. When adding videos to playlists the youtube app shows the playlists with chwckboxes & the boxes are checked next to the playlists the video is already added to... so it's easy to manage & prevents accidentally adding more than once.

& 

2. If users add too many external playlists newpipe glitches and the playlists default back to the 1st video in a playlist after each video watched in the playlist... perhaps adding the ability to switch between different profiles could be a useful feature to counter that glitch... currently the only way to bypass this is to export settings then delete playlists until the glitch stops and save the new settings and reimport initial settings to get the initial playlists back. Yet adding multiple profiles would do the same thing only more convieniently.

Otherwise switching to newpipe was fairly painless.

So for what it's worth, these are the issues current new users seem most likely to notice, imho.

**Comment by Stypox** [comment]
@ListSort could you open an issue for the bug 2.? Thank you!
As for 1., it is a far better solution to prevent duplicates than the one you proposed first. Could you edit this issue with that suggestion, or create a new one while closing this?

**Comment by illustribe** [comment]
+1! ;D

**Comment by amanharwara** [comment]
Sorting isn't really a good solution for avoiding duplicates, but it is a feature that would still be very useful. For e.g. being able to sort the videos by video title, video length, channel name, etc.

**Comment by matthew-piziak** [comment]
@amanharwara Agreed. I'm interested in this issue because I like watching dynamic playlists in ascending duration order—it shortens the playlist faster.

**Comment by SameenAhnaf** [comment]
>Sorting isn't really a good solution for avoiding duplicates

@amanharwara The author has already proposed a better solution in https://github.com/TeamNewPipe/NewPipe/issues/3263#issuecomment-605531838.

>When adding videos to playlists the youtube app shows the playlists with checkboxes & the boxes are checked next to the playlists the video is already added to... so it's easy to manage & prevents accidentally adding unwanted duplicates.

**Comment by Feuerswut** [comment]
yes, please

**Comment by drishal** [comment]
Man even i would love to see a sort playlist feature added, because i have made a watch later playlist on my youtube account and would like to sort by date added so i can later view the bookmarked playlist on the phone lol

**Comment by Monoxide8969** [comment]
Playlists need to have some way to be sorted by, bktg ascending and descending:

By dte of upload
By date of inclusion in the playlist
By title
By number of views

And I'm surrly missing some

**Comment by jlpalaciosb** [comment]
Playlists should be sorted by date descending and avoiding duplicates by default, i mean, why not?

**Comment by paisano117** [comment]
Hi, is this feature still being considered?

**Comment by Feuerswut** [comment]
> Hi, is this feature still being considered?

Currently, Newpipe is starting to get recoded, so that new features can be implemented without breaking old code.

This means, no new features will be added until the major components (player, UI, extractor, database) are in place and have proper internal APIs.

Upside is that new features will be implemented (such as modern material design 3) and also sorting features can be implemented faster, once the rewrite is done.

**Comment by matthew-piziak** [comment]
@paisano117 I'm using PocketTube in the meantime to sort YouTube playlists in place. Best wishes for the recode!
